### ISSUE TITLE
ZC1105 false positive: grouping parens flagged as nested arithmetic

### ISSUE THREAD

**Issue by afadesigns** [afadesigns/zshellcheck#1370]
ZC1105 ("Avoid nested arithmetic expansions") fires on arithmetic commands that contain only grouping parentheses, not a nested arithmetic expansion.

## Reproduce

```zsh
(( (i + 1) % n ))
(( (a + b) * (c + d) ))
(( result = (base + offset) * scale ))
```

All three are flagged, yet none contains a nested `$(( ... ))` expansion. They use ordinary precedence grouping `( ... )`.

## Cause

`checkZC1105` matches on the string form of the expression:

```go
exprString := arithCmd.Expression.String()
if strings.Contains(exprString, "$((") || strings.Contains(exprString, "((") { ... }
```

The AST printer wraps every infix and grouped sub-expression in parentheses, so a leading parenthesized operand renders as `((group) ...)`. The `((` substring then matches the printer output, not any source-level nesting.

## Impact

Grouping is the common, idiomatic way to control arithmetic precedence. Across the pinned corpora, the `((` branch produces the large majority of ZC1105 findings, and they are spurious. Severity is Style, so it is non-fatal, but the noise undermines trust in the rule.

## Fix

Drop the `|| strings.Contains(exprString, "((")` disjunct. Arithmetic grouping `( ... )` is not an expansion; only `$(( ... ))` is. The `$((` check alone still flags genuine nesting such as `(( x = $((y + 1)) ))`. A `(( ... ))` block cannot appear as a sub-expression inside another `(( ... ))` in valid Zsh (it is a command, not an expression), so the disjunct can only ever fire on grouping parens.
